osu.Game/Screens/Select/PlaySongSelect.cs

```diff
@@ -100,14 +109,50 @@ protected override bool OnExiting(Screen next)
             return false;
         }
 
+        protected override bool OnKeyDown(InputState state, KeyDownEventArgs args)
+        {
+            controlPressed = state.Keyboard.ControlPressed;
+            return base.OnKeyDown(state, args);
+        }
+
+        protected override bool OnKeyUp(InputState state, KeyUpEventArgs args)
+        {
+            controlPressed = state.Keyboard.ControlPressed;
+            return base.OnKeyUp(state, args);
+        }
+
         protected override void OnSelected()
         {
             if (player != null) return;
 
+            originalMods = modSelect.SelectedMods.Value;
+            if (controlPressed)
+                if (findAutoMod(originalMods) == null)
+                {
+                    var auto = findAutoMod(Ruleset.Value.CreateInstance().GetModsFor(ModType.Special));
+                    if (auto != null)
+                        modSelect.SelectedMods.Value = originalMods.Concat(new[] { auto });
+                }
+
             Beatmap.Value.Track.Looping = false;
             Beatmap.Disabled = true;
 
             LoadComponentAsync(player = new PlayerLoader(new Player()), l => Push(player));
         }
+
+        private Mod findAutoMod(IEnumerable<Mod> mods)
+        {
+            foreach (var mod in mods)
+            {
+                if (mod is ModAutoplay) return mod;
+                var multimod = mod as MultiMod;
+                if (multimod != null)
+                {
+                    var find = findAutoMod(multimod.Mods);
+                    if (find != null) return find;
+                }
+            }
+            return null;
+        }

```

Comment:
Why not just create a new virtual method as part of Ruleset to create the auto mod for you? I find this stuff pretty ugly.